code/modules/urist/modules/wfc/modules/deepmaint/generation.dm

```diff
@@ -0,0 +1,51 @@
+/proc/deepmaint_rebuild_mapfile(var/ruleset = "deepmaint.json")
+	var/success = generate_wfc_map(ruleset)
+	return success
+
+/area/map_template/deepmaint_wfc/proc/deepmaint_rebuild_mapfile(var/ruleset = "deepmaint.json")
+	var/success = generate_wfc_map(ruleset)
+	return success
```

Comment:
Are these procs actually necessary at all? `generate_wfc_map()` already returns a boolean, so these wrapper procs are adding nothing at all other than a name change. Can we not just call `generate_wfc_map(ruleset)` where necessary instead as there's no extra logic here?

Reply:
Not strictly needed, no. They are largely a legacy of a weird bug I've seen with the module-style includes.

For some reason callproc would flat out ignore 'functional' procs from module-included code but would recognize 'method' procs from it... so I bound it to the area.

Partially, I've also kept it in as an OOP-ish API into the underlying functional API, pytorch-style.

Reply:
iirc DM _sucks_ when it comes to proc calls and their related overhead. If you want to keep them that's fine, but for me it just feels like unnecessary clutter if you're circumventing the whole module system anyway

Reply:
It it, yeah, but even worst-case scenario, this is not on the hot path anywhere (except possibly periodic deepmaint rebuild, which is not even currently used by anything).

I would like to see what API people use more, initially.